src/server/gdb_server.c

```diff
@@ -3176,9 +3176,15 @@ static bool gdb_handle_vcont_packet(struct connection *connection, const char *p
 			return true;
 		}
 
-		retval = target_step(ct, current_pc, 0, 0);
-		if (retval == ERROR_TARGET_NOT_HALTED)
-			LOG_INFO("target %s was not halted when step was requested", target_name(ct));
+		if (ct->state == TARGET_UNAVAILABLE) {
+			LOG_TARGET_ERROR(ct, "Target is unavailable, so cannot be stepped. "
```

Comment:
You may consider whether to use just LOG_TARGET_WARNING() here, but I don't have a strong opinion.

Reply:
I'll leave this for now. We can change it if real-world use says this is too noisy.